Docs/reference/content/reference/driver/crud/compression.md

```diff
@@ -0,0 +1,108 @@
+
+### Specify compression via `URI`
+
+```c#
+using MongoDB.Driver;
+```
+
+To specify compression with [`ConnectionString`]({{<apiref "T_MongoDB_Driver_Core_Configuration_ConnectionString">}}), just add `compressors` into the connection string, as in:
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=snappy");
+var client = new MongoClient(mongoUrl);
+```
+for Snappy compression, or
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=zlib");
+var client = new MongoClient(mongoUrl);
+```
+for zlib compression, or 
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=zstd");
+var client = new MongoClient(mongoUrl);
+```
+for Zstandard compression, or 
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=snappy,zlib,zstd");
+var client = new MongoClient(mongoUrl);
+```
+to configure multiple compressors.
+
+In all cases the driver will use the first compressor in the list for which the server advertises support. 
+
+### Specify compression via `MongoClientSettings`
+
+```c#
+using MongoDB.Driver;
+using MongoDB.Driver.Core.Compression;
+using MongoDB.Driver.Core.Configuration;
+using System.Collections.Generic;
+```
+
+To specify compression with [`MongoClientSettings`]({{<apiref "T_MongoDB_Driver_MongoClientSettings">}}), set the `Compressors` property to a list of [`CompressorConfiguration`]({{<apiref "T_MongoDB_Driver_Core_Configuration_CompressorConfiguration">}}) instances:
+
+```c#
+var mongoClientSettings = new MongoClientSettings();
+mongoClientSettings.Compressors = new List<CompressorConfiguration>()
+{
+	new CompressorConfiguration(CompressorType.Snappy)
+};
+var client = new MongoClient(mongoClientSettings);
+```
+for Snappy compression, or
+
+```c#
+var mongoClientSettings = new MongoClientSettings();
+mongoClientSettings.Compressors = new List<CompressorConfiguration>()
+{
+	new CompressorConfiguration(CompressorType.Zlib)
+};
+var client = new MongoClient(mongoClientSettings);
+```
+for zlib compression, or
+
+```c#
+var mongoClientSettings = new MongoClientSettings();
+mongoClientSettings.Compressors = new List<CompressorConfiguration>()
+{
+	new CompressorConfiguration(CompressorType.Zstd)
+};
+var client = new MongoClient(mongoClientSettings);
+```
+for Zstandard compression, or
+
+```c#
+var mongoClientSettings = new MongoClientSettings();
+mongoClientSettings.Compressors = new List<CompressorConfiguration>()
+{
+	new CompressorConfiguration(CompressorType.Snappy),
+	new CompressorConfiguration(CompressorType.Zlib),
+	new CompressorConfiguration(CompressorType.Zstd)
+};
+var client = new MongoClient(mongoClientSettings);
+```
+to configure multiple compressors. 
+
+As with configuration with a URI, the driver will use the first compressor in the list for which the server advertises support.
```

Comment:
Nit: It sounds funny when `with` is used twice in a row.  How about:

    As with configuration via connection string